### PR TITLE
Ensure Read/Show instances are always provided with UTCTime and Universa...

### PR DESCRIPTION
**Pull request by gbaz**
As per the discussion on tickets #7 and #25 this should make sure that when Data.Time.Clock is imported (providing UTCTime and Universal) both the Read and Show instances of those data types also come into scope. I just did this by playing around with which module imports which, and in so doing created no cycles. The one migration of code was moving getCurrentTime from Data.Time.Clock (which re-exports it) to Data.Time.Clock.POSIX (so that it is available to Data.Time.LocalTime).

So the only visible difference to end-users, outside of better behavior overall, is that Data.Time.Clock.POSIX now provides an additional function that was previously only available via importing Data.Time.Clock. A further refactor could remove that change as well, but this is the way to accomplish things with minimal change, and it seems straightforward enough to me.

### CODE REVIEW

**Comment by erikd** [comment]
+1 LGTM.

**Comment by AshleyYakeley** [comment]
There are about nine orphan instances in this package I think. Can you do them all?

**Comment by gbaz** [comment]
This isn't about removing orphan instances. This is just about ensuring that Read/Show instances are, from the standpoint of the public API, always made available in the modules that provide the underlying data instances.

I don't think there were any other such cases in the package (this is to say, even if the instances are "internally" orphans, importing the public module from which you can get the data type also should pull in the read/show instances.).

If you can point me to other instances of this issue I'd be happy to take a look.

**Comment by bergmark** [comment]
Yes please! This has been a source of confusion for me on multiple occasions.

**Comment by AshleyYakeley** [comment]
@gbaz there are nine such cases altogether. Search for the phrase "instance of" to find them.

**Comment by gbaz** [comment]
@AshleyYakeley as I've mentioned, I don't know of any other such orphans where they are not made available to the user in importing the module to get the type (because these orphans are always "pulled in" to the higher level module which is actually exposed to users -- time, as you well know, is structured to have hidden and exposed modules both).

I'd be willing to look for more, but I would appreciate being pointed to another example of the issue I raised.

In the meantime, since this change is an improvement regardless, could it just be merged?

**Comment by AshleyYakeley** [comment]
@gbaz for example, Day is provided in Data.Time.Calendar, but its Read instance is not. There are nine in total, and they are mentioned in the haddock comments.

**Comment by AshleyYakeley** [comment]
Given the possible complexity of the change to the module imports, I would prefer they all be done together if possible.

**Comment by gbaz** [comment]
Gotcha -- grepping "instance of" I get "Day", "LocalTime", "ZonedTime", "TimeOfDay", and "TimeZone"

That's five, plus the two I got, gives seven. Where are the two I missed, or does that cover it?

**Comment by gbaz** [comment]
ok, this commit should nail the additional instances I listed -- it would be good to confirm, but again it was very straightforward.

File: lib/Data/Time/Calendar.hs [hunk 1]
```diff
@@ -6,3 +6,4 @@ module Data.Time.Calendar
 
 import Data.Time.Calendar.Days
 import Data.Time.Calendar.Gregorian
+import Data.Time.Format()
```
Comment (AshleyYakeley) [inline lib/Data/Time/Calendar.hs]: add missing newline...

Comment (gbaz) [inline lib/Data/Time/Calendar.hs]: fixed

**Comment by gbaz** [comment]
@AshleyYakeley any chance of this PR getting accepted now that I've added the other instances?

**Comment by AshleyYakeley** [comment]
It works in code, but the instances don't show up in haddock for some reason.

**Comment by gbaz** [comment]
Great! thanks!

**Pull request merged**